state/state.go

```diff
@@ -38,9 +38,11 @@ type State struct {
 	mtx sync.Mutex
 	db  dbm.DB
 
-	// should not change
-	GenesisDoc *types.GenesisDoc
-	ChainID    string
+	// See https://github.com/tendermint/tendermint/issues/671.
+	ChainID string
+	// consensus parameters used for validating blocks
+	// must not be nil
+	Params *types.ConsensusParams
```

Comment:
Do we need the pointer ?

Reply:
GenesisDoc uses a pointer value https://github.com/tendermint/tendermint/pull/676/files/2eba353e2ed3da4c1ae18b4bde3e7f853a7a66a3#diff-f6be34387a141f652301c16cf7a5e36cL272 so that's just transferring the field in here. However, to answer your question, usually I'll go with pointers because then I can find out if something was set or not idiomatically with
```go
if g.Params == nil {
}
```
as opposed to
```go
if g.Params == types.ConsensusParams{} {
}
```
or 
```go
if reflect.DeepEqual(g.Params, types.ConsensusParams{}) {
}
```

Reply:
I think it depends on how often params will be changing. If not very often, we can make it a raw struct. If the opposite, then making it a pointer makes sense for faster updates.